exchanges/ftx/ftx_wrapper.go

```diff
@@ -1254,3 +1266,179 @@ func (f *FTX) GetAvailableTransferChains(ctx context.Context, cryptocurrency cur
+		result.Price = decimal.NewFromFloat(info.Positions[i].Cost)
+		return result, nil
+	}
+	// order no longer active, use offline calculation
+	calc := order.PNLCalculator{}
+	result, err = calc.CalculatePNL(ctx, pnl)
+	if err != nil {
+		return nil, fmt.Errorf("%s %s %w", f.Name, f.API.Credentials.Subaccount, err)
+	}
+	return result, nil
+}
+
+// ScaleCollateral takes your totals and scales them according to FTX's rules
+func (f *FTX) ScaleCollateral(ctx context.Context, calc *order.CollateralCalculator) (decimal.Decimal, error) {
+	var result decimal.Decimal
+	if calc.CalculateOffline {
+		if calc.CollateralCurrency.Match(currency.USD) {
+			// FTX bases scales all collateral into USD amounts
+			return calc.CollateralAmount, nil
+		}
+		if calc.USDPrice.IsZero() {
+			return decimal.Zero, fmt.Errorf("%s %s %w to scale collateral", f.Name, calc.CollateralCurrency, order.ErrUSDValueRequired)
+		}
+		collateralWeight, ok := f.collateralWeight[calc.CollateralCurrency.Upper().String()]
+		if !ok {
+			return decimal.Zero, fmt.Errorf("%s %s %w", f.Name, calc.CollateralCurrency, errCollateralCurrencyNotFound)
+		}
+		if calc.CollateralAmount.IsPositive() {
+			if collateralWeight.InitialMarginFractionFactor == 0 {
+				return decimal.Zero, fmt.Errorf("%s %s %w", f.Name, calc.CollateralCurrency, errCollateralInitialMarginFractionMissing)
+			}
+			var scaling decimal.Decimal
+			if calc.IsLiquidating {
+				scaling = decimal.NewFromFloat(collateralWeight.Total)
+			} else {
+				scaling = decimal.NewFromFloat(collateralWeight.Initial)
+			}
+			weight := decimal.NewFromFloat(1.1 / (1 + collateralWeight.InitialMarginFractionFactor*math.Sqrt(calc.CollateralAmount.InexactFloat64())))
+			result = calc.CollateralAmount.Mul(calc.USDPrice).Mul(decimal.Min(scaling, weight))
+		} else {
+			result = result.Add(calc.CollateralAmount.Mul(calc.USDPrice))
+		}
+		return result, nil
+	}
+	wallet, err := f.GetCoins(ctx)
+	if err != nil {
+		return decimal.Zero, fmt.Errorf("%s %s %w", f.Name, calc.CollateralCurrency, err)
+	}
+	balances, err := f.GetBalances(ctx)
+	if err != nil {
+		return decimal.Zero, fmt.Errorf("%s %s %w", f.Name, calc.CollateralCurrency, err)
+	}
+	for i := range wallet {
+		if !currency.NewCode(wallet[i].ID).Match(calc.CollateralCurrency) {
+			continue
+		}
+		for j := range balances {
+			if !currency.NewCode(balances[j].Coin).Match(calc.CollateralCurrency) {
+				continue
+			}
+			scaled := wallet[i].CollateralWeight * balances[j].USDValue
+			result = decimal.NewFromFloat(scaled)
+			return result, nil
+		}
+	}
+	return decimal.Zero, fmt.Errorf("%s %s %w", f.Name, calc.CollateralCurrency, errCollateralCurrencyNotFound)
+}
+
+// CalculateTotalCollateral scales collateral and determines how much collateral you can use for positions
+func (f *FTX) CalculateTotalCollateral(ctx context.Context, collateralAssets []order.CollateralCalculator) (*order.TotalCollateralResponse, error) {
```

Comment:
Sub-accounts are mixing up with main account funds. We are calling 
[DEBUG]|REQUESTER| 14/01/2022 09:31:38 |FTX attempt 1 request path: https://ftx.com/api/wallet/balances multiple times and 
[DEBUG]|REQUESTER| 14/01/2022 09:31:38 |FTX attempt 1 request path: https://ftx.com/api/wallet/coins when I send the RPC request. 

Seems like there is no differentiation between balances and accounts and in the outgoing headers there is nothing to differentiate as well.

Reply:
I've added basic subaccount overriding in FTX now, this allows subaccount querying for retreiving collateral when calculating online.

I've also minimised the number of calls